### PR TITLE
[Merged by Bors] - chore: update for leanprover/std4#758

### PR DESCRIPTION
**Pull request by urkud**
---

- [x] depends on: #leanprover/std4#758

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/List/Rotate.lean [hunk 1]
```diff
@@ -592,14 +592,15 @@ theorem get_cyclicPermutations (l : List α) (n : Fin (length (cyclicPermutation
 #align list.nth_le_cyclic_permutations List.get_cyclicPermutations
 
 @[simp]
-theorem head?_cyclicPermutations (l : List α) : (cyclicPermutations l).head? = l := by
+theorem head_cyclicPermutations (l : List α) :
+    (cyclicPermutations l).head (cyclicPermutations_ne_nil l) = l := by
   have h : 0 < length (cyclicPermutations l) := length_pos_of_ne_nil (cyclicPermutations_ne_nil _)
-  simp_rw [← get_zero h, get_cyclicPermutations, rotate_zero]
+  rw [← get_mk_zero h, get_cyclicPermutations, Fin.val_mk, rotate_zero]
+  -- rw [← Option.some_inj, ← head?_eq_head, head?_cyclicPermutations]
```
Comment (grunweg) [inline Mathlib/Data/List/Rotate.lean:599]: Commented code.

**Review by grunweg — APPROVED** [review]
Looks good to me; thank you for doing this. (Pending the std PR and a std bump, of course.)

File: Mathlib/Data/List/Rotate.lean (hunk 1 above)
Comment (urkud) [inline Mathlib/Data/List/Rotate.lean:599]:
```suggestion
```

**Comment by kmill** [comment]
bors r-

(requested at https://leanprover.zulipchat.com/#narrow/stream/345428-mathlib-reviewers/topic/bors.20fail/near/436232802)

**Comment by semorrison** [comment]
bors p=5

**Comment by semorrison** [comment]
bors merge

**Pull request closed without merging**